### DISCUSSION TITLE
Issue with inclusion merge

### DISCUSSION THREAD

**Discussion by anamitrasikdar** [mtex-toolbox/mtex#1611] [Ask Anything]
Hi,
I am using merge.m function in the @grain2d folder of mtex-5.8.1. 

While I am running this line
**[grainsMerged, parentId] = merge(grains,'inclusions', 'maxSize', 10);**

I am getting the error,
### Error in grain2d/merge (line 101)
    isIncl(isIncl) = cond(grains.id2ind(hostId(isIncl)));

I am not able to figure out what is causing this error...

And also I want to understand why the line in the merge.m script runs without any problem whereas I am having trouble when I am calling "merge" from the script. Can anybody please look into the issue??

**Comment by kilir** [comment]
Hi,
could you also provide the relevant lines of your script, before you call merge.
Cheers,
Rüdiger

**Reply by anamitrasikdar** [reply]
### Thank you for replying. I copied some of the lines of the script. Here they are...


[grains,ebsd.grainId] = calcGrains(ebsd, 'angle', [hagb lagb],'boundary', 'tight');
ebsd(grains.grainSize <5 )) = [];
F = halfQuadraticFilter
F.alpha = 0.25  
ebsd_smooth = smooth(ebsd,F,'fill');
[grains_smooth1,ebsd_smooth.grainId] = calcGrains(ebsd_smooth,'angle', [hagb lagb],...
        'boundary', 'tight', 'removeQuadruplepoints');
 grains_smooth = grains_smooth1.smooth(5);
[grainsMerged, parentId] = merge(grains_smooth,'inclusions', 'maxSize', 10);

**Comment by anamitrasikdar** [comment]
Does anyone know what's wrong with the script I shared?
A quick look is highly appreciated.

Regards,
Anamitra

**Reply by ralfHielscher** [reply]
Hi Anamitra,

could you please save your grains variable as a mat file and then send them to us?

Thank you,
Ralf.

**Reply by anamitrasikdar** [reply]
Hi Prof. Hielscher,

Thank you very much for taking an interest.

I have zipped grains_smooth.mat and attached it here.

[grains_smooth.zip](https://github.com/mtex-toolbox/mtex/files/10011329/grains_smooth.zip)


Regards,
Anamitra

**Reply by kilir** [reply]
Hi,
using the current development version, those grains merge fine.
Cheers,
Rüdiger

**Reply by anamitrasikdar** [reply]
Hi Prof. Kilian,

I have attached a snap of the screen. Merge is working for pixel size 1 and 2, not beyond that. But I have inclusions with grain size definitely more than 2. 

Am I missing something here? 

And I am using 5.8.1.

Regards,
Anamitra

![Screen](https://user-images.githubusercontent.com/99597265/201934107-197db077-f41a-4c5c-94c8-33a246fd71c4.png)

**Reply by kilir** [reply]
I don't find a number where it does not work. But there have been a few changes since 5.8.1.

**Reply by anamitrasikdar** [reply]
Hi,
I am very happy to share that development version is working fine.
Thank you Prof. Kilian for the suggestion.

Is there any way we can update the version by calling from the MATLAB command window?

Thank you.

Regards,
Anamitra

**Reply by ralfHielscher** [reply]
I just uploaded MTEX 5.8.2. Currently, we have no way to update version from within command line.

Ralf.

**Reply by anamitrasikdar** [reply]
Prof. Hielscher,

Thank you for that.

Anamitra
